### PR TITLE
feat(DocumentDB): Add new DocumentDB check for cluster snapshot visibility

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

To strengthen data security and adhere to best practices for database management, we are introducing a new check within the Amazon DocumentDB framework. This check ensures that manual cluster snapshots are not publicly accessible.

### Description

I added `documentdb_cluster_snapshot_not_public` with his respective unit test. In order to do that I have added `ClusterSnapshot` model to `documentdb_service.py` with his respective unit test. 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "critical",
+  "ResourceType": "AwsRDSDBClusterSnapshot",
+  "Description": "Check if DocumentDB manual cluster snapshot is public.",
+  "Risk": "If you share an unencrypted manual snapshot as public, the snapshot is available to all AWS accounts. Public snapshots may result in unintended data exposure.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-3",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws docdb modify-db-snapshot-attribute --db-snapshot-identifier <snapshot_id> --attribute-name restore --values-to-remove all",
+      "NativeIaC": "",
+      "Other": "",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json]:
```suggestion
      "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-3",
```

File: prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "CheckTitle": "Check if DocumentDB manual cluster snapshot is public.",
+  "CheckType": [],
+  "ServiceName": "documentdb",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+  "Severity": "critical",
+  "ResourceType": "AwsRDSDBClusterSnapshot",
+  "Description": "Check if DocumentDB manual cluster snapshot is public.",
+  "Risk": "If you share an unencrypted manual snapshot as public, the snapshot is available to all AWS accounts. Public snapshots may result in unintended data exposure.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-3",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json]: Please put a related URL from the AWS docs.

File: prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "Remediation": {
+    "Code": {
+      "CLI": "aws docdb modify-db-snapshot-attribute --db-snapshot-identifier <snapshot_id> --attribute-name restore --values-to-remove all",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "To remove public access from a manual snapshot, follow the Sharing a snapshot tutorial.",
+      "Url": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-3"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json]: Please put a related URL from the AWS docs.

File: prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "documentdb_cluster_snapshot_not_public",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_snapshot_not_public/documentdb_cluster_snapshot_not_public.metadata.json]:
```suggestion
  "CheckID": "documentdb_cluster_public_snapshot",
```

**Comment by MarioRgzLpz** [comment]
@sergargar rename check and urls added in commit bb97c40

**Review by MrCloudSec — APPROVED** [review]
Great job 👏🏼

**Pull request merged**